### PR TITLE
Adding samples to SDK pipeline

### PR DESCRIPTION
**Pull request by patilsnr**

### CODE REVIEW

File: vsts/gatedBuild.ps1 [hunk 1]
```diff
@@ -58,7 +58,7 @@ else
 {
 	#Likely a nightly or CI build
 	Write-Host "Not a pull request build, will run all tests"
```
Comment (drwill-ms) [inline vsts/gatedBuild.ps1:60]:
```suggestion
	Write-Host "Not a pull request build, will run all tests and samples"
```

File: vsts/vsts.yaml [hunk 2]
```diff
@@ -466,12 +479,17 @@ jobs:
       - powershell: ./vsts/gatedBuild.ps1
         displayName: build
         env:
-          # Environment variables for IoT Hub E2E tests
+                   # Environment variables for IoT Hub E2E tests
```
Comment (drwill-ms) [inline vsts/vsts.yaml:482]: Fix tabbing

File: vsts/vsts.yaml [hunk 3]
```diff
@@ -494,7 +512,7 @@ jobs:
           E2E_IKEY: $(E2E-IKEY)
 
           # Environment variables for the DevOps pipeline
-          PROXY_SERVER_ADDRESS: 127.0.0.1:3128
+          PROXY_SERVER_ADDRESS: 127.0.0.1:8888
```
Comment (drwill-ms) [inline vsts/vsts.yaml:515]: I think we have the proxy set in a keyvault setting, but I could be wrong. Should we pull it out of there?

**Review by drwill-ms — APPROVED** [review]

File: vsts/vsts.yaml (hunk 3 above)
Comment (abhipsaMisra) [reply vsts/vsts.yaml:515]: you'll want to check the proxy server address for both linux (docker image) and windows (squid proxy setup). If they're the same then we can put the value in keyvault. If they're different then we should hardcode it in our yaml directly and add a note explaining why.

Comment (andyk-ms) [reply vsts/vsts.yaml:511]: Linux and windows have different proxy and they have different ports.
For windows, we install SQUID and use the default setup. The proxy port should be 3128.
If indeed we want to change this port, squid/squid.conf need to setup with this port number.

File: azureiot.sln [hunk 4]
```diff
@@ -162,40 +162,40 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SimulatedDevice", "iothub\d
...
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ComputeDerivedSymmetricKeySample", "provisioning\device\samples\Getting Started\ComputeDerivedSymmetricKeySample\ComputeDerivedSymmetricKeySample.csproj", "{A2D72CEF-7CB6-439D-A875-22DB39899969}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ComputeDerivedSymmetricKeySample", "provisioning\device\samples\getting started\ComputeDerivedSymmetricKeySample\ComputeDerivedSymmetricKeySample.csproj", "{A2D72CEF-7CB6-439D-A875-22DB39899969}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "samples", "samples", "{8F9D9145-B786-4A2C-9A59-E1C33E07F5A2}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "X509Sample", "provisioning\device\samples\Getting Started\X509Sample\X509Sample.csproj", "{CAD27793-E63B-4579-B6A8-31FAA9AC8BE4}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "X509Sample", "provisioning\device\samples\getting started\X509Sample\X509Sample.csproj", "{CAD27793-E63B-4579-B6A8-31FAA9AC8BE4}"
```
Comment (patilsnr) [inline azureiot.sln:169]: Fix inconsistencies with directory naming, e.g. 'how to`, `How To`, `how to guides`. All folder names are now lowercase and level 2 samples are all in directories named `how to guides`

File: build.ps1 [hunk 5]
```diff
@@ -70,7 +71,8 @@ Param(
     [string] $framework = "*",
     [switch] $skipIotHubTests,
     [switch] $skipDPSTests,
-    [switch] $noBuildBeforeTesting
+    [switch] $noBuildBeforeTesting,
+    [switch] $runSamples
```
Comment (patilsnr) [inline build.ps1:75]: add a new function to the build script that builds and runs all samples as part of nightly runs

File: vsts/vsts.yaml [hunk 6]
```diff
@@ -313,7 +318,7 @@ jobs:
           E2E_IKEY: $(E2E-IKEY)
 
           # Environment variables for the DevOps pipeline
-          PROXY_SERVER_ADDRESS: 127.0.0.1:8888
+          PROXY_SERVER_ADDRESS: 127.0.0.1:3128
```
Comment (schoims) [inline vsts/vsts.yaml:321]: port needs to be updated [here](https://github.com/Azure/azure-iot-sdk-csharp/blob/ac3e71036edd0a099c5c7283628085a252d4c01a/vsts/vsts.yaml#L284)?

**Pull request merged**